packages/react-internal/src/components/Modal/Modal.styles.ts

```diff
@@ -30,6 +31,9 @@ export const getStyles = (props: IModalStyleProps): IModalStyles => {
 
   const classNames = getGlobalClassNames(globalClassNames, theme);
 
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const win = useWindow();
```

Comment:
instead of calling the hook here, we should add `windowInnerHeight?: number` style prop and pass `win?.innerHeight` from component.